web/README.md

```diff
@@ -36,4 +43,52 @@ First, ensure you are on the `release` branch. Next, run:
 $ GIT_USER=<Your GitHub username> USE_SSH=true npm run deploy
 ```
 
-This command will build the website and push it to the `gh-pages` branch.
+This command will build the website and push it to the `gh-pages` branch,
+which will get it deployed to https://wasp-lang.dev !
+
+### Multiple documentation versions
+
+We maintain docs for multiple versions of Wasp.
+
+Docusaurus docs on this: https://docusaurus.io/docs/versioning .
+
+Docusaurus recognizes "current" docs, which are docs in ./docs dir, and also
+individual versioned docs, which are docs under versioned_docs/{version}/.
+So we have 1 "current" docs and a number of versioned docs.
+
+We stick with Docusaurus' recommended/default approach for multiple doc versions, which says that "current" docs, which reside in `docs/` dir and are served under `/docs/next` URL, are work in progress (even though they are named "current", which is a bit misleading).
+So "current" docs are docs for the version of Wasp that is currently in development, not for the last released version, and they are not meant to be consumed by typical Wasp user.
+
+Each versioned documentation consists of versioned_docs/{version} and
+versioned_sidebars/{version}.
+There is also versions.json file which is just a list of versioned docs.
+
+By default, "current" docs are served under URL {baseUrl}/docs/next,
+each versioned doc is served under URL {baseUrl}/docs/{version},
+and last versioned docs (first version in versions.json)
+are served under URL {baseUrl}/docs/, as the default/latest docs.
+
+Since we don't want our users to read `docs/next` ("current" docs), we don't publish these when we deploy docs, instead we build them only during development.
+
+#### When/how do we create new version of docs from "current" docs?
+
+When releasing new version of Wasp, what we do is run `npm run docusaurus docs:version {version}` to create new versioned docs from the current docs. We do this on every new Wasp release.
```

Comment:
@Martinsos Maybe we want to do this for each **major** version to keep the number of published docs lower?

So the version is `0.11.X` and not `0.11.8` and `0.12.0` and `0.12.1`

Reply:
I explained live why doing it in such way is not so easy. Approach I proposed is the best fit for our current dynamics. However, what we can do is delete older versions, for example we can delete 0.12.1 if we published 0.12.2. We can even play with labels/paths to achieve having 0.12 all the time, as you suggested, that is also possible, but it would really just be us creating new version of docs, removing previous one, and naming this new one the same way. So the approach wouldn't change, but we can maneuver around it to achieve this if needed.